stargate/pkg/auth/client.go

```diff
@@ -34,6 +37,11 @@ type authRequest struct {
 	Password string `json:"password"`
 }
 
+var cacheEnabled = true
```

Comment:
Do you think these can be variables of an instance of `tableBasedTokenProvider`?

Reply:
Sensible. I'm not totally sure I got it right, but I think I'm close. Works, in any case.